updated with the latest changes in t3-oss/create-t3-turbo

trevorpfiz opened the pull request:
## What kind of change does this PR introduce?

updated to the latest changes in [t3-oss/create-t3-turbo](https://github.com/t3-oss/create-t3-turbo)

## What is the current behavior?

was using Expo Router v2 and Expo SDK 49.

## What is the new behavior?

Should be pretty much the same, but using Expo Router v3 and Expo SDK 50 as implemented in t3-oss/create-t3-turbo.

## Additional context

didn't spend much time on the light/dark mode colors

trevorpfiz committed:
update all but app

trevorpfiz committed:
missed some things, working on expo app and nextjs

trevorpfiz committed:
expo app updated

trevorpfiz committed:
updated nextjs app, ready to test

trevorpfiz committed:
updated repo to latest create-t3-turbo changes, tested that GitHub login works on web and Apple on Expo Go

trevorpfiz commented:
I will look into replacing `auth-helpers` with `@supabase/ssr`

trevorpfiz committed:
update with latest changes

trevorpfiz committed:
Merge pull request #1 from trevorpfiz/04-01-update_with_latest_changes

update with latest changes

kyh commented:
🚢

trevorpfiz committed:
changed next.js from auth-helpers to supabase/ssr

trevorpfiz committed:
Merge pull request #2 from trevorpfiz/04-02-changed_next.js_from_auth-helpers_to_supabase_ssr

changed next.js from auth-helpers to supabase/ssr

trevorpfiz commented:
updated the Next.js app to `@supabase/ssr`, but not sure of the best way to handle the Expo app.

sourdzl commented:
thank you so much for this update!

trevorpfiz committed:
updated with latest changes

trevorpfiz committed:
Merge pull request #3 from trevorpfiz/05-07-update

updated with latest changes

zocario commented:
Hello @trevorpfiz thanks for the work on this pull request 🙏  I'm starting a new project and want to use the t3-turbo template with Supabase, but as this repository is not up-to-date I'm wondering if your fork is ready to be used for starting a new project? Do you know your PR will be merged soon?

trevorpfiz committed:
updated readme for migrations and trigger

trevorpfiz committed:
updated readme for migrations and trigger (#4)

trevorpfiz commented:
@zocario it should be ready for starting a new project. let me know if you run into any issues or have any questions as you are building.

as for the PR being merged, it should be ready, but I still have some questions about `@supabase/ssr` and the Expo app. SDK 51 just got released as well.

I am also still figuring out the best pattern for using Drizzle migrations with Supabase for local development/branching.

trevorpfiz committed:
drizzle-zod

trevorpfiz committed:
drizzle-zod (#5)

trevorpfiz committed:
forgot to add profile.id reference

trevorpfiz committed:
forgot to add profile.id reference (#6)

trevorpfiz committed:
fixed migrate script

trevorpfiz committed:
fixed migrate script (#7)

limegorilla commented:
Hi @trevorpfiz - seem to be running into issues when using this with a local Supabase instance. @Vercel/postgres doesnt seem to like the connection URL I am using - any tips?

Currently have tried the standard one provided:
```bash
postgresql://postgres:postgres@127.0.0.1:54322/postgres
postgresql://postgres:postgres@localhost:54322/postgres /// @vercel/postgres does a check to see if localhost is in the name and drops some of it's checks
```

If this isn't supported please let me know - I have tried using the `node-postgres` and `pg` clients but they all seem to have their own problems with webpack bundling.

trevorpfiz commented:
@limegorilla I will play around with local Supabase and let you know.

zocario commented:
> @zocario it should be ready for starting a new project. let me know if you run into any issues or have any questions as you are building.
> 
> as for the PR being merged, it should be ready, but I still have some questions about `@supabase/ssr` and the Expo app. SDK 51 just got released as well.
> 
> I am also still figuring out the best pattern for using Drizzle migrations with Supabase for local development/branching.

Thanks! I've been trying the template and there is something I'm not sure to understand:
- When I run `pnpm dev` or `pnpm --filter expo dev` I have those errors with the environment variables of the expo Apps (I've added values in `.env` file in `apps/expo`: 
```
Error: Error reading Expo config at /Users/enzo/Developer/t3-supabase-test/apps/expo/app.config.ts:       
                                                                                                                        
Please provide SUPABASE_URL and SUPABASE_ANON_KEY in your .env file                                                     
Error: Error reading Expo config at /Users/enzo/Developer/t3-supabase-test/apps/expo/app.config.ts:
```
- If I run `pnpm --filter nextjs dev` I'm able to run nextjs and it seems to be working. If I run `npx expo start` I don't have environment variables errors, but when I open an iOS simulator I get this error:
```
iOS Bundling failed 12ms (node_modules/expo/AppEntry.js)
Unable to resolve "../../App" from "node_modules/expo/AppEntry.js"
```

So I'm a little bit confused on what command to use to successfully launch the Expo apps and if using `pnpm dev` to launch everything as in the original t3-turbo repository is still possible?

I've followed those steps to start my project:
1. Use your project as template
2. Run `pnpm i`
3. Add .env file in root folder with `POSTGRES_URL`, `NEXT_PUBLIC_SUPABASE_URL`, `NEXT_PUBLIC_SUPABASE_ANON_KEY`
4. Add .env file in `apps/expo` with `EXPO_PUBLIC_SUPABASE_URL`, `EXPO_PUBLIC_SUPABASE_ANON_KEY`.

zocario commented:
Also @trevorpfiz I noticed that when running `pnpm db:push` I had an error due to `author_id` being a varchar instead of uuid, modifying it by `authorId: uuid("author_id")` fixes the issue on my side.

Edit: doing that then lead to error when creating a post, sounds like there is more work to do or I went in the wrong direction after the `db:push` error 👀 
 
```
>>> tRPC Error on 'post.create'  [TRPCError: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "undefined",
    "path": [
      "authorId"
    ],
    "message": "Required"
  }
]] {
  code: 'BAD_REQUEST',
  name: 'TRPCError'
}
```

trevorpfiz commented:
@zocario Thanks for the info. I will polish things up later this weekend and let you know what I find.

trevorpfiz committed:
fix lint errors, update drizzle-kit

trevorpfiz committed:
fix lint errors, update drizzle-kit (#8)

trevorpfiz committed:
chore: Expo 51 & RN 0.74

trevorpfiz committed:
chore: Expo 51 & RN 0.74 (#9)

trevorpfiz commented:
@zocario the rough edges should be fixed. 

@limegorilla still have to get to trying Supabase locally. does it work if you don't use `@vercel/postgres`? https://orm.drizzle.team/learn/tutorials/drizzle-with-supabase

limegorilla commented:
> @limegorilla still have to get to trying Supabase locally. does it work if you don't use `@vercel/postgres`? https://orm.drizzle.team/learn/tutorials/drizzle-with-supabase

Played around with this at the weekend - `node-postgres` has issues with Webpack bundling in this configuration and you have to patch Webpack to get it to ignore the extensions for Cloudflare and Neon I believe.

Managed to get the `postgres` driver working however and that seems quite happy. There is the option to conditionally use `postgres` locally and `@vercel/postgres` when in production but to me that is going to create problems down the line when your environments are that separate.

I do plan on putting together a mirror with this change made as a default at some point.

zocario commented:
> @zocario the rough edges should be fixed.
> 
> @limegorilla still have to get to trying Supabase locally. does it work if you don't use `@vercel/postgres`? https://orm.drizzle.team/learn/tutorials/drizzle-with-supabase

I tested this morning and everything is looking good, thank you! 👍 

I noticed one small thing: the email in the `profile` table isn't filled after signup (using the trigger on authentication) but maybe this is a problem in my configuration, not a blocking issue anyway :)

trevorpfiz committed:
feat: multi-root workspaces

trevorpfiz committed:
feat: multi-root workspaces (#10)

only issue is that the launch.json at the root of the workspace is redundant when working in multi-root workspace. not sure how to support single and multi-root.

trevorpfiz committed:
feat: protected routes

trevorpfiz committed:
feat: protected routes (#11)

trevorpfiz committed:
feat: new signin pages, light mode works

trevorpfiz committed:
feat: new signin pages, light mode works (#12)

using middleware to secure routes
auth/signin and auth/signup pages
next-safe-action
validation schemas
Light mode works

trevorpfiz commented:
> > @zocario the rough edges should be fixed.
> > @limegorilla still have to get to trying Supabase locally. does it work if you don't use `@vercel/postgres`? https://orm.drizzle.team/learn/tutorials/drizzle-with-supabase
> 
> I tested this morning and everything is looking good, thank you! 👍
> 
> I noticed one small thing: the email in the `profile` table isn't filled after signup (using the trigger on authentication) but maybe this is a problem in my configuration, not a blocking issue anyway :)

yeah I am seeing the issue with email signups. must not have the info when the trigger runs before the email is verified? will look into a fix.

williamneves commented:
Hi, I'm trying you PR with supbase and I'm getting this errors:

```js
import { createEnv } from "@t3-oss/env-core";
import postgres from 'postgres'
import { sql } from "@vercel/postgres";
import { drizzle } from "drizzle-orm/vercel-postgres";
import { z } from "zod";

import * as schema from "./schema";

export const env = createEnv({
  server: {
    POSTGRES_URL: z.string(),
  },
  // eslint-disable-next-line no-restricted-properties
  runtimeEnv: process.env,
  emptyStringAsUndefined: true,
});


const client = postgres(env.POSTGRES_URL, {prepare: false})

export const db = drizzle(sql, { schema });
```

`Error: VercelPostgresError - 'invalid_connection_string': This connection string is meant to be used with a direct connection. Make sure to use a pooled connection string or try `createClient()` instead.`

I tried with supabase local and remote, same error

I tried to remove the vercel and put the postgress-js, and get another error about cloudflare-workers, I need to disable the edge run time to work.

Did you also face this issue?

trevorpfiz committed:
fix: no email on signup

trevorpfiz committed:
fix: no email on signup (#13)

This pull request updates the Supabase email templates and redirect URLs. It also adds a trigger on user verification. Additionally, adds a check if user tries to sign up with email but already has an account.

---

trevorpfiz commented:
> Hi, I'm trying you PR with supbase and I'm getting this errors:
> 
> ```js
> import { createEnv } from "@t3-oss/env-core";
> import postgres from 'postgres'
> import { sql } from "@vercel/postgres";
> import { drizzle } from "drizzle-orm/vercel-postgres";
> import { z } from "zod";
> 
> import * as schema from "./schema";
> 
> export const env = createEnv({
>   server: {
>     POSTGRES_URL: z.string(),
>   },
>   // eslint-disable-next-line no-restricted-properties
>   runtimeEnv: process.env,
>   emptyStringAsUndefined: true,
> });
> 
> 
> const client = postgres(env.POSTGRES_URL, {prepare: false})
> 
> export const db = drizzle(sql, { schema });
> ```
> 
> `Error: VercelPostgresError - 'invalid_connection_string': This connection string is meant to be used with a direct connection. Make sure to use a pooled connection string or try `createClient()` instead.`
> 
> I tried with supabase local and remote, same error
> 
> I tried to remove the vercel and put the postgress-js, and get another error about cloudflare-workers, I need to disable the edge run time to work.
> 
> Did you also face this issue?

@williamneves Maybe double check that your `POSTGRES_URL` is in this format, `POSTGRES_URL="postgres://postgres.[USERNAME]:[PASSWORD]@aws-0-eu-central-1.pooler.supabase.com:6543/postgres?workaround=supabase-pooler.vercel"`? Can also try the nonPoolingUrl which should maybe not be commented out in `drizzle.config.ts`. Also, `@vercel/postgres` is needed instead of `postgres` if you want to use edge.

trevorpfiz commented:
> > @zocario the rough edges should be fixed.
> > @limegorilla still have to get to trying Supabase locally. does it work if you don't use `@vercel/postgres`? https://orm.drizzle.team/learn/tutorials/drizzle-with-supabase
> 
> I tested this morning and everything is looking good, thank you! 👍
> 
> I noticed one small thing: the email in the `profile` table isn't filled after signup (using the trigger on authentication) but maybe this is a problem in my configuration, not a blocking issue anyway :)

I believe just changing the email on the trigger to `new.email` instead of `new.raw_user_meta_data ->> 'email'` fixed it?

williamneves commented:
> 6543

Yeah, that `?workaround=supabase-pooler.vercel` solves. Now I saw on the env.example have an example like that.

Is it a good idea to explicitly this on documentation; –).

Huge thanks; I loved this repo!

trevorpfiz committed:
fix: nonPoolingUrl, export alias, update drizzle-kit

trevorpfiz committed:
Merge pull request #16 from trevorpfiz/05-15-fix_nonpoolingurl_export_alias_update_drizzle-kit

fix: nonPoolingUrl, export alias, update drizzle-kit

trevorpfiz commented:
@juliusmarminge I think this is ready to merge, I'm interested to hear your thoughts. Also, what is the vision for this fork now that `create-t3-turbo` is using Supabase? the Supabase Auth fork?

Here are the differences from `main` in this PR:
1. `@supabase/ssr` for `nextjs`
2. added sign-in and sign-up pages in `nextjs` to better support password login
3. Drizzle migrations
4. `.code-workspace` file for multi-root workspaces option
5. README additions

WIP not in this PR:
1. Best way to handle Expo auth with the `auth-helpers` going into maintenance mode?
2. `react-native-reusables` for UI components, light/dark mode, etc. in `expo` (have been running into `nativewind` bugs)

trevorpfiz commented:
Figured out that `"react-native-css-interop": "~0.0.34"` was the culprit for the nativewind bugs. Remove or use `"react-native-css-interop": "~0.0.36"`.

Will look into adding some features from `react-native-reusables` soon

rickylabs commented:
Are there any blockers preventing the merge of this PR?

ringge commented:
is this PR going to be merged soon?
Thanks for all the hard works

ulzhang commented:
bump. this has just been sitting here

BenjaOliva commented:
Hope this get's merged soon. Had to make almost the same changes in order to get development build working and updated packages

ThingEngineer commented:
This Expo React Native Starter has been on the Supabase home page for so long. Any update one when/if this PR will be merged?
